### PR TITLE
fork the pull-kubernetes-bazel-build|test jobs have simpler target lists on master

### PR DESCRIPTION
**Pull request by BenTheElder**
…arget lists while master just has //...

### CODE REVIEW

File: prow/config.yaml [hunk 1]
```diff
@@ -1236,16 +1339,18 @@ presubmits:
     rerun_command: "/test pull-kubernetes-verify"
     trigger: "(?m)^/test( all| pull-kubernetes-verify),?(\\s+|$)"
   kubernetes-security/kubernetes:
-  - name: pull-security-kubernetes-bazel-build
+  - name: pull-kubernetes-bazel-build
```
Comment (krzyzacy) [inline prow/config.yaml]: keep the `-security` in name

Comment (BenTheElder) [reply prow/config.yaml]: fixed

**Comment by krzyzacy** [comment]
@ixdy shall we also `bazel test //...` as well?

**Comment by ixdy** [comment]
`+212 −6` ugh.

and yes, we need to fork the -test jobs too.

**Comment by BenTheElder** [comment]
/hold cancel
did -test

File: prow/config.yaml [hunk 2]
```diff
@@ -1210,6 +1363,9 @@ presubmits:
           mountPath: /root/.cache
         ports:
         - containerPort: 9999
+Writing objects: 100% (4/4), 680 bytes | 680.00 KiB/s, done.
```
Comment (BenTheElder) [inline prow/config.yaml]: ??? not sure how this happened, fixing.

Comment (BenTheElder) [reply prow/config.yaml]: fixed..

**Comment by BenTheElder** [comment]
Back to green, this should work but if/when we merge it we need to watch it :fire:

**Comment by krzyzacy** [comment]
(I'll let @ixdy stamp)

**Comment by BenTheElder** [comment]
Should be good to go?

File: prow/config.yaml [hunk 3]
```diff
@@ -167,7 +169,7 @@ presubmits:
         - "--service-account=/etc/service-account/service-account.json"
         - "--upload=gs://kubernetes-jenkins/pr-logs"
         - "--" # end bootstrap args, scenario args below
-        - "--build=//cmd/... //pkg/... //federation/... //plugin/... //third_party/... //examples/... //test/... //vendor/k8s.io/..."
+        - "--build=//..."
```
Comment (ixdy) [inline prow/config.yaml]: master is currently broken, so we shouldn't merge this.

**Comment by ixdy** [comment]
I still maintain that copy-pasting this is the wrong way to do this.

**Comment by BenTheElder** [comment]
> I still maintain that copy-pasting this is the wrong way to do this.

Sure but for now it should work and we don't seem to have a consensus on an alternative yet.

File: prow/config.yaml (hunk 3 above)
Comment (BenTheElder) [reply prow/config.yaml]: I dropped a /hold until https://github.com/kubernetes/kubernetes/pull/53681#issuecomment-336566466 is sorted

**Comment by BenTheElder** [comment]
/hold
should be good once https://github.com/kubernetes/kubernetes/pull/53915 merges

**Comment by BenTheElder** [comment]
That merged, but we should hold off until next week when we are around to babysit things in case of :fire: :-)

**Comment by krzyzacy** [comment]
/assign @ixdy 
Jeff seems has most of the concerns

**Comment by ixdy** [comment]
/lgtm

I'm still unhappy about this.

File: prow/config.yaml (hunk 3 above)
Comment (ixdy) [reply prow/config.yaml]: we need to support negative targets in the scenario; `make bazel-build` is `bazel build -- //... -//vendor/...`.

File: prow/config.yaml [hunk 4]
```diff
@@ -353,7 +456,54 @@ presubmits:
...
+      - image: gcr.io/k8s-testimages/bazelbuild:v20171005-2f37588f-0.6.1
+        args:
+        - "--clean"
+        - "--git-cache=/root/.cache/git"
+        - "--job=$(JOB_NAME)"
+        - "--repo=k8s.io/$(REPO_NAME)=$(PULL_REFS)"
+        - "--service-account=/etc/service-account/service-account.json"
+        - "--upload=gs://kubernetes-jenkins/pr-logs"
+        - "--" # end bootstrap args, scenario args below
+        - "--test=//..."
```
Comment (ixdy) [inline prow/config.yaml]: similarly `make bazel-test` is `//... //hack:verify-all -//build/... -//vendor/...`

Comment (BenTheElder) [reply prow/config.yaml]: I'll update this once we have negative deps in.

File: prow/config.yaml (hunk 3 above)
Comment (BenTheElder) [reply prow/config.yaml]: done

File: prow/config.yaml (hunk 4 above)
Comment (BenTheElder) [reply prow/config.yaml]: done

**Comment by BenTheElder** [comment]
this should be good to go pending on #5026

**Comment by BenTheElder** [comment]
/hold
(until #5026)

**Comment by BenTheElder** [comment]
/hold cancel
(#5026 was merged)

**Comment by krzyzacy** [comment]
negative deps is in
/lgtm
/hold

**Comment by BenTheElder** [comment]
/hold cancel
will watch this which should be fine, we've been soaking negative deps for a while now succesfully, I will revert if anything goes sideways somehow :fire:

**Comment by BenTheElder** [comment]
tested in: https://github.com/kubernetes/kubernetes/pull/46662#issuecomment-338079510 with an unrelated bazel-test failure (flake?)

**Comment by BenTheElder** [comment]
not unrelated, see https://github.com/kubernetes/test-infra/pull/5090
need to filter out e2e.

**Comment by BenTheElder** [comment]
to recap:
- needed to filter e2e as well as integration
- needed to manually exclude `//examples/...` which is broken and tagged as manual

**Pull request merged**